downloadDirectory/src/main/java/org/reactome/release/downloadDirectory/GenerateGOAnnotationFile/BiologicalProcessAnnotationBuilder.java

```diff
@@ -0,0 +1,122 @@
+        if (catalystInstances.size() == 0) {
+            Set<GKInstance> proteinInstances = GOAGeneratorUtilities.retrieveProteins(reactionInst);
+            processProteins(proteinInstances, reactionInst);
+        } else {
+            // Check that catalyst instance has no disqualifying attributes.
+            for (GKInstance catalystInst : catalystInstances) {
+                GKInstance catalystPEInst = (GKInstance) catalystInst.getAttributeValue(ReactomeJavaConstants.physicalEntity);
+                boolean validCatalyst = GOAGeneratorUtilities.validateCatalyst(catalystPEInst);
+                if (validCatalyst) {
+                    Set<GKInstance> proteinInstances = getBiologicalProcessProteins((GKInstance) catalystInst.getAttributeValue(ReactomeJavaConstants.physicalEntity));
+                    processProteins(proteinInstances, reactionInst);
+                }
+            }
+        }
+    }
+
+    // Returns all protein instances from multi-instance PhysicalEntitys (Polymer/Complex/EntitySet) or just the single incoming instance if it is an EWAS.
+    private static Set<GKInstance> getBiologicalProcessProteins(GKInstance physicalEntityInst) throws Exception {
+        Set<GKInstance> proteinInstances = new HashSet<>();
+        if (GOAGeneratorUtilities.multiInstancePhysicalEntity(physicalEntityInst.getSchemClass())) {
+            proteinInstances.addAll(getMultiInstanceSubInstances(physicalEntityInst));
+        } else if (physicalEntityInst.getSchemClass().isa(ReactomeJavaConstants.EntityWithAccessionedSequence)) {
+            proteinInstances.add(physicalEntityInst);
+        }
+        return proteinInstances;
+    }
+
+    // Attempt to generate GO Annotation information for each protein associated with the whole Reaction or just it's catalysts, depending on the goTerm being evaluated.
+    private static void processProteins(Set<GKInstance> proteinInstances, GKInstance reactionInst) throws Exception {
+
+        for (GKInstance proteinInst : proteinInstances) {
+            GKInstance referenceEntityInst = (GKInstance) proteinInst.getAttributeValue(ReactomeJavaConstants.referenceEntity);
+            GKInstance speciesInst = (GKInstance) proteinInst.getAttributeValue(ReactomeJavaConstants.species);
+            // Check if the protein has any disqualifying attributes.
+            boolean validProtein = GOAGeneratorUtilities.validateProtein(referenceEntityInst, speciesInst);
+            if (validProtein) {
+                String taxonIdentifier = ((GKInstance) speciesInst.getAttributeValue(ReactomeJavaConstants.crossReference)).getAttributeValue(ReactomeJavaConstants.identifier).toString();
+                // TODO: Rule check
+                if (!microbialSpeciesToExclude.contains(taxonIdentifier)) {
+                    getGOBiologicalProcessLine(referenceEntityInst, reactionInst, taxonIdentifier);
```

Comment:
Does the line returned need to captured and returned so it gets back to the main class for the output file?